### PR TITLE
Modeling

### PR DESCRIPTION
**Pull request by danparnella**
### Items Addressed
- Initial pass at generating models with factories and tests
- Getting everything to build and deploy correctly

### CODE REVIEW

File: Gemfile [hunk 1]
```diff
@@ -10,7 +10,6 @@ gem 'rubocop-performance'
 gem 'rubocop'
 gem 'rails_util', github: 'launchpadlab/rails_util'
 gem 'rack-cors', '~> 1.0'
-gem 'paper_trail', '~> 10.0'
```
Comment (danparnella) [inline Gemfile:13]: Figured I was using this and was receiving a log error so I just removed it.

File: Gemfile [hunk 2]
```diff
@@ -5,12 +5,11 @@ ruby '2.7.0'
 
 gem 'rails', '6.1.3.1'
 gem 'token_master', '~> 1.0'
-gem 'sentry-raven', '~> 2.0'
```
Comment (danparnella) [inline Gemfile:8]: Removed as it required additional set up and I won't be using it.

File: Gemfile [hunk 3]
```diff
@@ -5,12 +5,11 @@ ruby '2.7.0'
...
 gem 'rubocop-performance'
 gem 'rubocop'
 gem 'rails_util', github: 'launchpadlab/rails_util'
 gem 'rack-cors', '~> 1.0'
-gem 'paper_trail', '~> 10.0'
+gem 'paper_trail', '~> 11.0'
```
Comment (danparnella) [inline Gemfile:12]: Upgraded to remove console warning about v10 not being compatible with Rails 6.1

File: Gemfile [hunk 4]
```diff
@@ -27,12 +26,15 @@ gem 'webpacker', '~> 5.0'
 
 gem 'turbolinks', '~> 5'
 
+# newly added
+gem 'credit_card_validations'
```
Comment (danparnella) [inline Gemfile:30]: Added a credit card validation gem

File: Gemfile [hunk 5]
```diff
@@ -27,12 +26,15 @@ gem 'webpacker', '~> 5.0'
...
+
 
 
 gem 'bootsnap', '>= 1.4.4', require: false
 
 group :development, :test do
-  gem 'rspec-rails', '4.0.0.beta3'
+  gem 'rspec-rails', '~> 4.0'
```
Comment (danparnella) [inline Gemfile:37]: This upgraded `rspec-rails` which removed a `name` error for every test when running `rspec`. [[Reference](https://stackoverflow.com/questions/65637389/ruby-on-rails-failure-error-on-all-examples)]

File: Gemfile [hunk 6]
```diff
@@ -56,4 +58,4 @@ group :test do
   gem 'shoulda-matchers', '~> 3.1'
 end
 
-gem 'tzinfo-data', platforms: [:mingw, :mswin, :x64_mingw, :jruby]
+# gem 'tzinfo-data', platforms: [:mingw, :mswin, :x64_mingw, :jruby]
```
Comment (danparnella) [inline Gemfile:61]: Commenting out since I'm not using Windows and it caused a warning in the console logs.

File: Thoughts.md [hunk 7]
```diff
@@ -0,0 +1,43 @@
...
+    - Most popular times of the day to see a movie
+    - Total revenue
+    - Revenue by day of the week
+- Site must work on mobile
+- Capture thought processes in Readme file
+- Push code to GitHub and delpoy to Heroku (provide both links when finished)
+
+- Created [DB Diagram](https://dbdiagram.io/d/60a2e757b29a09603d154743)
+
+- Not sure if a Customer should have orders or if an Order should have customers.
```
Comment (danparnella) [inline Thoughts.md:43]: Realized a Customer would have multiple orders so an Order should belong to a Customer.

File: app/models/customer.rb [hunk 8]
```diff
@@ -0,0 +1,8 @@
+class Customer < ApplicationRecord
+  VALID_EMAIL_REGEX = /\A[\w+\-.]+@[a-z\d\-]+(\.[a-z\d\-]+)*\.[a-z]+\z/i
```
Comment (danparnella) [inline app/models/customer.rb:2]: Ripped from the Learn Enough Rails tutorial

File: db/seeds.rb [hunk 9]
```diff
@@ -6,4 +6,3 @@
 #   movies = Movie.create([{ name: 'Star Wars' }, { name: 'Lord of the Rings' }])
 #   Character.create(name: 'Luke', movie: movies.first)
 
-Seeds::UserSeeder.new.run
```
Comment (danparnella) [inline db/seeds.rb:9]: `Seeds` was not defined when running `db:seed` so I removed this and plan on just entering my seed commands directly in this file.

File: Gemfile (hunk 3 above)
Comment (inveterateliterate) [reply Gemfile:12]: https://github.com/LaunchPadLab/rails_template/issues/124 😁

File: Gemfile (hunk 5 above)
Comment (inveterateliterate) [reply Gemfile:37]: do we need to update the rails template for this too?

**Pull request merged**